Core/src/IfcGeometryConverter/GeomUtils.cpp

```diff
@@ -1043,6 +1043,29 @@ void GeomUtils::convertPlane2Matrix( const carve::geom::vector<3>& plane_normal,
 	resulting_matrix._44 = 1;
 }
 
+/**********************************************************************************************/
+/*! \brief Applies a position to the point.
+*
+* \param[in] listOfPoint				A list of points to be converted.
+* \param[in] positionMatrix			A position matrix, which should be applied to the points.
+* \param[in] ammountOfPoints			Number of points.
+*
+* \return[out]							List of points with applied position.
+*/
+static void  applyPositionToVertex(
+	const std::vector<carve::geom::vector<3>>& listOfPoints,
+	const carve::math::Matrix & positionMatrix,
+	const int ammountOfPoints,
+	std::vector<carve::geom::vector<3>>& newListOfPoints
+) noexcept(false) {
+	for (int i = 0; i < ammountOfPoints; ++i)
+	{
+		carve::geom::vector<3> vertex = listOfPoints.at(i);
+		vertex = positionMatrix * vertex;
+		newListOfPoints.push_back(vertex);
+	}

```

Comment:
Simpler code:

```suggestion
	for ( const auto& vertex : listOfPoints )
	{
		newListOfPoints.push_back(positionMatrix * vertex);
	}
```

Therefore, no need for the amount of points being passed.

Doxycommnt names of variables do not match.

Reply:
Additionally: if the size of the array is known upfront, it is advisable to reserve the amount of space in advance (speed optimisation). Prepend the for loop with

    newListOfPoints.reserve( listOfPoints.size() );

Reply:
Resolved: 3f70103